localstack-core/localstack/services/lambda_/invocation/executor_endpoint.py

```diff
@@ -214,3 +218,66 @@ def invoke(self, payload: Dict[str, str]) -> InvocationResult:
             invoke_timeout_buffer_seconds = 5
             timeout_seconds = lambda_max_timeout_seconds + invoke_timeout_buffer_seconds
         return self.invocation_future.result(timeout=timeout_seconds)
+
+    @staticmethod
+    def _perform_invoke(
+        invocation_url: str,
+        proxies: dict[str, str],
+        payload: dict[str, Any],
+    ) -> requests.Response:
+        """
+        Dispatches a Lambda invocation request to the specified container endpoint, with automatic
+        retries in case of connection errors, using exponential backoff.
+
+        The first attempt is made immediately. If it fails, exponential backoff is applied with
+        retry intervals starting at 100ms, doubling each time for up to 5 total retries.
+
+        Parameters:
+            invocation_url (str): The full URL of the container's invocation endpoint.
+            proxies (dict[str, str]): Proxy settings to be used for the HTTP request.
+            payload (dict[str, Any]): The JSON payload to send to the container.
+
+        Returns:
+            Response: The successful HTTP response from the container.
+
+        Raises:
+            requests.exceptions.ConnectionError: If all retry attempts fail to connect.
+        """
+        backoff = None
+        last_exception = None
+        max_retry_on_connection_error = 5
+
+        for attempt_count in range(max_retry_on_connection_error + 1):  # 1 initial + n retries
+            try:
+                response = requests.post(url=invocation_url, json=payload, proxies=proxies)
+                response.raise_for_status()
```

Comment:
The response is checked on line 200 as well - this will prevent the other error handling, and leak request errors to the callers (instead of the expected InvokeSendError). We should not raise for the status here, I think the connection error should still be raised, right?

Reply:
Yes indeed I could confirm; I removed this raise in the invoke routine